api.planx.uk/modules/analytics/metabase/shared/client.ts

```diff
@@ -0,0 +1,155 @@
+import axios from "axios";
+import type {
+  AxiosInstance,
+  AxiosError,
+  AxiosRequestConfig,
+  InternalAxiosRequestConfig,
+} from "axios";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+// Custom error class for Metabase-specific errors
+export class MetabaseError extends Error {
+  constructor(
+    message: string,
+    public statusCode?: number,
+    public response?: unknown,
+  ) {
+    super(message);
+    this.name = "MetabaseError";
+  }
+}
+
+interface MetabaseConfig {
+  baseURL: string;
+  apiKey: string;
+  timeout?: number;
+  retries?: number;
+}
+
+// Validate environment variables
+const validateConfig = (): MetabaseConfig => {
+  const baseURL = process.env.METABASE_BASE_URL;
+  const apiKey = process.env.METABASE_API_KEY;
```

Comment:
We'll need to add both of these env vars to the codebase. This can be done as part of this PR, or another one - whichever's best for you. It will need to merge either before or alongside this one though ideally 👍 

There's a guide on this here - https://github.com/theopensystemslab/planx-new/blob/main/doc/how-to/how-to-add-a-secret.md

Happy to talk this through sometime also if required.

Reply:
Good shout, (hopefully!) handled on #3992